lib/cosmos/top_level.rb

```diff
@@ -642,7 +643,7 @@ def self.kill_thread(owner, thread, graceful_timeout = 1, timeout_interval = 0.0
       end
       if thread.alive?
         # Graceful failed
-        Logger.warn "Failed to gracefully kill thread:\n  #{thread.backtrace.join("\n  ")}\n"
+        Logger.warn "Failed to gracefully kill thread:\n  Caller Backtrace:\n  #{caller().join("\n  ")}\n  \n  Thread Backtrace:\n  #{thread.backtrace.join("\n  ")}\n\n"
```

Comment:
Added some more information to help debug threads not shutting down gracefully.  Not shows what the thread was doing and what thread tried to kill it.